src/analysis/check_checker/mod.rs

```diff
@@ -70,12 +133,14 @@ impl<'a, 'b> CheckChecker<'a, 'b> {
     fn new(
         db: &'a mut AnalysisDatabase<'b>,
         annotations: &'a Vec<Annotation>,
+        settings: Settings,
     ) -> CheckChecker<'a, 'b> {
         Self {
             db,
+            settings,
             taint_sources: HashMap::new(),
             tainted_nodes: HashMap::new(),
-            diagnostics: Vec::new(),
+            diagnostics: HashMap::new(),
```

Comment:
Should we go with a `BTreeMap`, to make the traversals more handy when using the diagnostics?

Reply:
We don't actually care about the order by the key though. The usage is only for inserts, removes, and then ultimately flattening it to a vector (below). Would a BTreeMap be more efficient for that flattening part?

```rust
        // Collect all of the vecs of diagnostics into a vector
        let mut diagnostics: Vec<Vec<Diagnostic>> = self.diagnostics.into_values().collect();
        // Order the sets by the span of the error (the first diagnostic)
        diagnostics.sort_by(|a, b| a[0].spans[0].cmp(&b[0].spans[0]));
        // Then flatten into one vector
        Ok(diagnostics.into_iter().flatten().collect())
```